src/test/java/com/napier/sem/DAO_IntegrationTests.java

```diff
@@ -279,4 +279,62 @@ void topNCitiesIn_allAreasMatchFilter ()
             // then
             for (City city : cities) assertEquals("United Kingdom", city.getCountry());
         }
+
+        /**
+         * Integration tests covering the DAO.allCapitalCitiesIn method
+         */
+        // Tests that all capital cities in query are in the given area
+        @Test
+        void allCapitalCitiesIn_allAreasMatchFilter()
+        {
+            // given
+            String areaFilter = "Brazil";
+
+            // when
+            ArrayList<CapitalCity> capitalCities = dao.allCapitalCitiesIn(App.COUNTRY, areaFilter);
+
+            // then
+            for (CapitalCity capitalCity: capitalCities) assertEquals(areaFilter, capitalCity.getCountry());
+        }
+
+        // Tests that an empty array is returned when the areaName is set incorrectly
+        @Test
+        void allCapitalCitiesIn_arrayIsEmptyWhenAreaNameIsInvalid() {
+            // given
+            String areaName = "East Fife";
+
+            // when
+            ArrayList<CapitalCity> capitalCities = dao.allCapitalCitiesIn(App.CONTINENT, areaName);
+
+            // then
+            assertEquals(0, capitalCities.size());
+        }
+
+        // Tests that when querying for capital cities in a country, no more than one is returned
+        @Test
+        void allCapitalCitiesIn_countryDoesNotHaveMoreThanOnceCapitalCity() {
+            // given
+            String areaName = "Germany";
+
+            // when
+            ArrayList<CapitalCity> capitalCities = dao.allCapitalCitiesIn(App.COUNTRY, areaName);
+
+            // then
+            assertTrue(capitalCities.size() <= 1);
+        }
+
+        // Tests that the returned results are correct
+        @Test
+        void allCapitalCitiesIn_resultCapitalCitiesAreExpected ()
+        {
+            // given
+            String expectedCapitalCity = "Roma";
+
+            // when
+            ArrayList<CapitalCity> capitalCities = dao.allCapitalCitiesIn(App.COUNTRY, "Italy");
```

Comment:
same for this one, i'd say we should test it on a small region, i'm not sure of an example of a small one off the top of my head though.